core/backend/src/ECSqlStatement.ts

```diff
@@ -332,46 +336,43 @@ export class ECSqlStatement implements IterableIterator<any>, IDisposable {
    * - [Code Samples]($docs/learning/backend/ECSQLCodeExamples#working-with-the-query-result)
    */
   public getRow(args?: ECSqlRowArg): any {
-    args = args ?? {};
+    if (!this._stmt)
+      throw new Error("ECSqlStatement is not prepared");
 
-    // Set default rowFormat if not provided
+    args = args ?? {};
     if (args.rowFormat === undefined) {
       args.rowFormat = QueryRowFormat.UseJsPropertyNames;
     }
-
-    const toRowOptions = {
-      classIdsToClassNames: true,
-      rowFormat: args.rowFormat,
-    };
-    const resp = this._stmt!.toRow(toRowOptions); // eslint-disable-line @typescript-eslint/no-non-null-assertion
+    if (args.classIdsToClassNames === undefined) {
+      args.classIdsToClassNames = true;
+    }
+    const resp = this._stmt.toRow({
+      classIdsToClassNames: args.classIdsToClassNames,
+      useJsName: args.rowFormat === QueryRowFormat.UseJsPropertyNames,
+      abbreviateBlobs: false,
+      // on 5.x this should be removed or set to false. It is only here to maintain compatibility with 4.x
+      doNotConvertClassIdsToClassNamesWhenAliased: true,
```

Comment:
Does 'whenAliased' mean if I were to `select classId as aliasedClassId from table` ?

Reply:
It seems that I'm wrong since you ended up changing some tests from `assert.isTrue(Id64.isValidId64(row.myParentRelClassId));` to `assert.isFalse(Id64.isValidId64(row.myParentRelClassId));`

Wouldnt the above be an aliased class Id and therefore it should stay as a classId?

Reply:
You are not wrong. It's a part I left to do for today. For some reason, the name reported when we convert to json is ECRelClassId instead of myParentRelClassId. I was going to debug this one to find why is that case.

Reply:
So I look into it and it expected. I was confused at first as well. But the result is from two different API. I first aligned output of both API `ECSqlStatement.toRow()` & `ECSqlReader`.

Then I added `doNotConvertClassIdsToClassNamesWhenAliased` flag for `ECSqlStatement` and set it to true and then updated the test. 

Following is set for ECSqlReader which is standard behavior. ECSqlStatement has different behavior.  On 5.x we will ensure both have same behavior by remove `doNotConvertClassIdsToClassNamesWhenAliased` or setting it to `false`

`assert.isTrue(Id64.isValidId64(row.myParentRelClassId));` to `assert.isFalse(Id64.isValidId64(row.myParentRelClassId));`

Reply:
So if I understand correctly...

ECSqlReader by default converts classIds to classNames (I guess because they are generally more useful to read than an ID)?. ECSQLStatement used to not convert classIds to classNames but it recently started to in some recent PR which caused the regression and broke some queries? Thats the reason for this option, to temporarily make that behavior opt-in in the case of ECSQLStatement. 

What was the motivation behind moving ECSQLStatement to start converting classIds to classNames in that previous PR? I'm guessing to unify the outputs of the Reader and the Statement( seems like a good reason).

Reply:
> What was the motivation behind moving ECSQLStatement to start converting classIds to classNames in that previous PR? I'm guessing to unify the outputs of the Reader and the Statement( seems like a good reason).
That’s right, we’ve transitioned from using TypeScript base values to object serialization, allowing us to leverage common code on the native side. Additionally, the performance of toRow() has significantly improved, so you’ll notice it’s much faster now.

The aliased property to retrieve the actual ID has been a bug since itwin.js 1.x. Users who need the ID rather than the className find this particularly frustrating, as there’s no way to bypass the default conversion from classID to className other than aliasing it. Some users have resorted to joining the ECClassDef class to get the ID, while others have used casting methods.

This is why I’m completely against this smart conversion. It might benefit a few, but it frustrates the majority. If someone wants the className, they should use SELECT ec_classname(ECClassId). We shouldn’t apply this conversion by default. We should return exactly what the user specifies in their query.

If they want ECInstanceId to be id, they should use `SELECT ECInstanceId AS id`. For camel case, they can simply alias the properties as needed.

Hopefully on 5.x we will need no flags at all.

Reply:
Totally agree with not doing smart conversion. While reading the PR I was also confused by the ECIsnstanceId->Id conversions happening seemingly magically. It also seemed to happen for SourceECInstnaceId, and even any properties with the name ECClassId in them as well.

At the very least, any and all smart conversion we do should be documented somewhere for people to read. Is that already the case?
When to use ECSqlStatement vs ECSqlReader also seems worth documenting if that isn't already. 

Doesn't have to happen in this PR, but if we could atleast have workitems that would be nice

Reply:
Yes, I intend to write a git issue for this explaining the issue and direction we should take on 5.x